### ISSUE TITLE
GUI transactions list (in the wallet overview) shows different descriptions for identical transactions (minor issue)

### ISSUE THREAD

**Issue by seanPhill** [Veil-Project/veil#988]
Noticed while running a script doing identical ringct to ringct transfers from one wallet to another.

The receiving wallet's transactions list in the Qt GUI wallet shows differently for no apparent reason.
These are all for the identical sent amount 50000.00000000 veil, all from the RingCT balance.

If the amount showing as received appears as 49999.99999999 (that is, one satoshi short) then the description will invariably say "Received with RingCT Address anonymous". If the received amount shows as 50000.00000000 then the description will say "Received with RingCT Address stealth". 

This is minor, but might be addressed in the long run as part of an overhaul of what this list displays with code inherited from the Bitcoin Core Qt wallet. Other displayed transactions that we have gotten used to are more confusing for new users, such as how we immediately know what a transaction is, just from the value of the negative amount shown, such as the fee divided by 0.01 is the number of zerocoins minted, while the description may vary depending on whether the mint occurred from basecoin and where the change went.

**Comment by Zannick** [comment]
Each transaction is supposed to be 2 txos, right? The UI should add them together to get the original send. I wonder if 1 sat actually got taken out or if it misunderstood the split somehow. Weird how that seems to be consistent with the address not being recognized.

**Comment by seanPhill** [comment]
Well, here's an example of one showing 1 satoshi missing in the wallet, and "Received with RingCT Address anonymous".
Two RingCT outputs and a RangeProof is correct. 
TXID dd69220d8b29d439a11d38f0d2d2c712c53e56f449c4545d477ce217f9a026be

**Comment by seanPhill** [comment]
When I look at get transaction ... in the wallet I can see the one satoshi, which reports as `"is_mine": false,` but the one that shows as `"is_mine": true,` shows the entire value without losing the satoshi, and the first part of vout is
```
        "type": "data",
        "ct_fee": "0.0037272",
        "data": "06f0df16",
        "is_mine": false,
        "has_tx_rec": true
```
the fee that was paid by the sending wallet.

**Comment by us77ipis** [comment]
@seanPhill You could check if this issue is also solved by #1001

**Comment by seanPhill** [comment]
Yes, partially, and in my opinion acceptably! It seems a little idiosyncractic, but there may be a hidden reason for the label change between "stealth" and "anonymous". PR1001 fixed all the amounts, which now show as exactly 50,000 VEIL for all of these identical CT to RingCT transactions I sent to another fresh wallet.
![PR1001-Fixes-Shown-Amount](https://user-images.githubusercontent.com/46495592/172637155-cbfb3cde-ae46-40c3-b243-6025bd7dfe50.png)

**Comment by seanPhill** [comment]
The above was filtered to show only RingCT transactions. The first, and in one case, the second also, in a series of these 50,000 Veil amounts received in RingCT were labelled "Stealth", while subsequent ones received shortly later were labelled "Anonymous". In the notifications of the operating system it mentioned that it was "Label: Stealth", while the subsequent ones were received _without a label._ I have no issue with this behaviour and (as the original reporter of the issue) recommend that we close this issue.

**Comment by seanPhill** [comment]
By the way, this is as observed on testnet. It is very unlikely that any significant number of people could duplicate or test or care about this remaining issue of some (CT to RingCT transactions) received labelled "stealth" and some not labelled, on mainnet.
